cyclonedx-bom/src/models/metadata.rs

```diff
@@ -41,6 +44,15 @@ pub struct Metadata {
 }
 
 impl Metadata {
+    /// Constructs a new `Metadata` with a timestamp of DateTime::now()
+    /// ```
+    /// use cyclonedx_bom::models::metadata::Metadata;
+    ///
+    /// let metadata = Metadata::new();
```

Comment:
This isn't a very good example since this returns a result, so open to feedback here. In `generator.rs` we are using the `?` operator but can't do that in the doctest as it's currently written.

Reply:
Went with this style: https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html#using--in-doc-tests